src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8622,7 +8647,29 @@ internal static List<AlternateStreamData> GetStreams(string path)
             SafeFindHandle handle = NativeMethods.FindFirstStreamW(
                 path, NativeMethods.StreamInfoLevels.FindStreamInfoStandard,
                 findStreamData, 0);
-            if (handle.IsInvalid) throw new Win32Exception();
+            if (handle.IsInvalid)
+            {
+                int error = Marshal.GetLastWin32Error();
+
+                // Directories don't normally have alternate streams, so this is not an exceptional state.
+                // If a directory has no alternate data streams, FindFirstStreamW returns ERROR_HANDLE_EOF.
+                if (error == NativeMethods.ERROR_HANDLE_EOF)
+                {
+                    return alternateStreams;
+                }
+
+                // Filesystems other than NTFS generally don't support alternate streams, and the system
+                // sets ERROR_INVALID_PARAMETER in that case.
+                if (error == NativeMethods.ERROR_INVALID_PARAMETER)
+                {
+                    throw new NotSupportedException(
+                        string.Format(FileSystemProviderStrings.AlternateDataStreamNotSupported, path));
+                }
```

Comment:
The PR is too complex already. I suggest to postpone this until we get a conclusion in related issue.

Reply:
You had suggested that this case be given a friendly error message, and then that the string be moved to the .resx for localization.  But okay, I'll back out this block and the .resx change to put in the related issue.